src/modules/systemlib/bson/tinybson.h

```diff
@@ -193,6 +193,17 @@ typedef struct bson_encoder_s {
  */
 __EXPORT int bson_encoder_init_file(bson_encoder_t encoder, int fd);
 
+/**
+ * Initialze the encoder for writing to a file.
+ *
+ * @param encoder		Encoder state structure to be initialised.
+ * @param fd			File to write to.
+ * @param buf			Buffer pointer to use
```

Comment:
Might be worth noting that `buf` cannot be null (opposed to `bson_encoder_init_buf`)

Reply:
done